packages/cli/lib/daemon/index.js

```diff
@@ -1,49 +1,60 @@
-import { WebSocketServer } from 'ws'
+// The one process that contains all networking resources, database, etc.
+import { Level } from 'level'
 
-import DHT from '@hyperswarm/dht'
-import { relay } from '@hyperswarm/dht-relay'
-import Stream from '@hyperswarm/dht-relay/ws'
+import { DEFAULT_PORT, REQUESTS, SEEDER_DATABASE_DIRECTORY } from '../constants.js'
+import Seeder from './seeder.js'
+import runRelay from './relay.js'
+import { respond } from '../utils.js'
 
-/**
- * @param {object} opts
- * @param {DHTOpts} [opts.dhtOpts]
- * @param {number} [opts.port]
- */
-export default async function run (opts = {}) {
-  const server = setupServer(opts.port)
-  const destroyDHT = dhtRelay(server, opts.dhtOpts)
-
-  return {
-    /** @type {number} */ // @ts-ignore
-    port: server.address().port,
-    close: () => Promise.all([destroyDHT(), server.close()])
-  }
-}
+const { dht, server } = runRelay({ port: DEFAULT_PORT })
 
-/** @param {number} [port] */
-function setupServer (port = 0) {
-  const server = new WebSocketServer({ port })
+server.on('connection', (socket) => {
+  socket.onmessage = onMessage
+})
 
-  server.on('error', () => {})
-  server.on('connection', socket => {
-    socket.on('error', () => {})
-  })
+const db = new Level(SEEDER_DATABASE_DIRECTORY, { valueEncoding: 'json' })
 
-  return server
-}
+const seederDB = db.sublevel('seeder')
+const seeder = new Seeder(dht, seederDB)
 
 /**
- * @param {import('ws').Server} server
- * @param {DHTOpts} [opts]
+ * @param {import('ws').MessageEvent} event
  */
-function dhtRelay (server, opts) {
-  const dht = new DHT(opts)
-
-  server.on('connection', socket => {
-    relay(dht, new Stream(false, socket))
-  })
+function onMessage (event) {
+  const request = tryParseJSON(event.data)
+  if (!request) return
+
+  switch (request.type) {
+    case REQUESTS.SEEDER_ADD:
+      seeder.add(request.payload.urls)
+        .then(() => {
+          respond(event.target, 'SEEDER_ADD', { status: 'ok' })
```

Comment:
I assume it was intended to use `RESPONSE.SEEDER_ADD` instead of `'SEEDER_ADD'`... But the intention of those constants is not clear to me.
However, it may make sense to have a map/object for matching `method`/`input`/`output` based on `request.type` and it will allow to avoid having this `switch` block all together and have something like this (in the most complicated case) `seeder[methods[request.type]](params[request.type]).then((res) => respond(event.target, request.type, response[request.type](res)))` instead

Reply:
The purpose is just to have a good DX, as LSP detect any typo, basically a JS enum.
Notice the JSDOC type definition of `type` param:
```js
/**
 * @param {import('ws').WebSocket} ws
 * @param {keyof import('./constants.js')['REQUESTS']} type
 * @param {object} payload
 */
export function request (ws, type, payload) {
  return ws.send(JSON.stringify({
    type: REQUESTS[type],
    payload
  }))
}

/**
 * @param {import('ws').WebSocket} ws
 * @param {keyof import('./constants.js')['RESPONSES']} type
 * @param {object} payload
 */
export function respond (ws, type, payload) {
  return ws.send(JSON.stringify({
    type: RESPONSES[type],
    payload
  }))
}
```

Reply:
No need to be too fancier in my opinion, just coping with not having Rust's enums.

Reply:
Not sure if speak about this same thing... Dev experience in case of adding new method in current implementation would mean editing request types + response types + onMesage + seeder = 4 places. Especially when you can have one place - `seeder`.